Source/WebCore/style/RuleData.h

```diff
@@ -86,6 +88,7 @@ class RuleData {
     unsigned m_linkMatchType : 2; //  SelectorChecker::LinkMatchMask
     unsigned m_propertyAllowlist : 2;
     unsigned m_isStartingStyle : 1;
+    unsigned m_isBaseAppearanceStyle : 1;
```

Comment:
I think we could avoid growing RuleData for this since it is mutually exclusive with starting-style (that is you can have m_isStartingOrBaseAppareanceStyle bit). There is never starting-style on UA sheet nor base appearance on non-UA sheet and I think all the clients know the current cascade level.

At some point we likely need to grow RuleData though. If perf tests come neutral I'd be fine with this too.